lib/fixup.rb

```diff
@@ -3,30 +3,52 @@
 require 'etc'
 require 'json'
 require_relative 'color'
-require_relative 'package'
-require_relative 'convenience_functions'
+
+def require_gem(gem_name_and_require = nil, require_override = nil)
+  # Allow only loading gems when needed.
+  return if gem_name_and_require.nil?
+
+  gem_name = gem_name_and_require.split('/')[0]
+  begin
+    gem gem_name
+  rescue LoadError
+    puts " -> install #{gem_name} gem".orange
+    Gem.install(gem_name)
+    gem gem_name
+  end
+  requires = if require_override.nil?
+               gem_name_and_require.split('/')[1].nil? ? gem_name_and_require.split('/')[0] : gem_name_and_require
+             else
+               require_override
+             end
+  require requires
+end
+require_gem('highline')
 
 # All needed constants & variables should be defined here in case they
 # have not yet been loaded or fixup is being run standalone.
 
-CREW_VERBOSE = ARGV.intersect?(%w[-v --verbose]) unless defined?(CREW_VERBOSE)
+CREW_VERBOSE ||= ARGV.intersect?(%w[-v --verbose])
 
-HOME = '/home/chronos/user' unless defined?(HOME)
-CREW_PREFIX = '/usr/local' unless defined?(CREW_PREFIX)
-CREW_LIB_PATH = File.join(CREW_PREFIX, 'lib/crew') unless defined?(CREW_LIB_PATH)
-CREW_CONFIG_PATH = File.join(CREW_PREFIX, 'etc/crew') unless defined?(CREW_CONFIG_PATH)
-CREW_META_PATH = File.join(CREW_CONFIG_PATH, 'meta') unless defined?(CREW_CONFIG_PATH)
+HOME ||= '/home/chronos/user'
```

Comment:
Why do we need to do this complex dance of conditional assignment for these variables? Why can't we just require const.rb?

Reply:
It's a good question!

We `load` fixup.rb... but since we already have const.rb required, as I understand it... it won't be loaded again unless we use `load`, which gives us a ton of `warning: already initialized constant ` errors.

The problem is that we NEED const.rb reloaded, because otherwise we will call remove.rb for a package that fixup wants to remove, but then remove.rb will die if crew update was run so long ago the in-memory const.rb doesn't have an essential_packages array constant loaded.

Reply:
I am still getting the warning: already initialized constant  errors though... Any suggestions on how to deal with this?

Reply:
Oh, that is tricky. Couple things: I've removed the need for essential_deps in #9832, so that's a possible solution.

In the short term, couldn't we load essential_deps from device.json in commands/remove.rb and bypass the need for it to be present in lib/const.rb?

Reply:
> I am still getting the warning: already initialized constant  errors though... Any suggestions on how to deal with this?

My suggestion is to take the warning at face value-- the constants are already initialised, so we don't need to set them again.

Reply:
Yes, but it looks really ugly... and we don't want to throw unnecessary warnings. :/

Reply:
Hence why I am suggesting not defining the constants again in lib/fixup.rb (i.e. why do we need to define HOME again?)

Reply:
HOME is no longer defined again.

Reply:
It's not just HOME-- why do we need to define most of these constants again, and why do we need to set all the constants in `lib/const.rb` as `||=` when we only redefine a few of them?

and again, all of this could be bypassed if we didn't load `essential_deps` from device.json

Reply:
You need the `||=` construction because we are using ` load 'lib/const.rb'` instead of using `require 'lib/const.rb'` to reload `const.rb` from `fixup.rb`, and that triggers tons of errors about trying to redefine constants.

(We reload const.rb from fixup.rb right after a git pull, so we may have new or changed constants there we need to check for.)

`||=` does a _define if not already defined_, which appears to work for most stuff, except anything we are defining with a call to check an environment variable, so that's mostly ideal for a file of constant definitions that we are about to reload.

Reply:
We could try a system call to `lib/fixup.rb` so its not dealing with any scope problems and loads everything afresh? (i.e. `system ./lib/fixup.rb` and just have it be a regular script)